core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/CombinedResponseHandler.java

```diff
@@ -68,6 +69,8 @@ public CombinedResponseHandler(HttpResponseHandler<OutputT> successResponseHandl
                                              ExecutionAttributes executionAttributes)
             throws IOException, InterruptedException {
 
+        logRequestId(httpResponse);
```

Comment:
Any specific reason to have picked this response handler to do the logging?

Reply:
This is where we log the response status code currently, so I just included the request ID here